react/finger-exercises/redux/02 - react-redux/src/app/screens/App/index.js

```diff
@@ -1,38 +1,40 @@
 import React, { Component, Fragment } from 'react';
-import store from '@redux/store';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
 import Navbar from '@components/Navbar';
 import Footer from '@components/Footer';
 
+import actionsCreators from '../../../redux/book/actions';
+
 import Book from './components/Book';
 import Search from './components/Search';
 import ShoppingCart from './components/ShoppingCart';
 import styles from './styles.scss';
 
 class App extends Component {
-  state = {
-    books: [],
-    bookSelected: []
-  };
-
   componentDidMount() {
-    store.subscribe(() => {
-      const { books, bookSelected } = store.getState();
-      this.setState({ books, bookSelected });
-    });
-    // TODO to implement the dispatch
+    this.props.getBooks();
   }
 
   // TODO to implement the dispatch
-  onSearch = value => {};
+  onSearch = value => {
+    this.props.searchBook(value);
```

Comment:
This method is just bypassing the value to the prop function. As you are not applying any extra logic to the value you may avoid defining this method and just pass the prop to the component. (The same for `addToCart `, `addItem ` and `removeItem `)
```jsx
const { books, bookSelected, onSearch, addToCart, addItem, removeItem } = this.props;
return (
  ...
  <Search onSearch={onSearch} />
  ...
  <ShoppingCart data={bookSelected} addItem={addItem} removeItem={removeItem} />
```
Also, i don't see `addToCart` being used.

Reply:
Excelent! I used "addToCart" into an objetct previusly to render. I should also change this definition for this.props.addToCart